libcaf_core/caf/uri_builder.cpp

```diff
@@ -18,7 +18,13 @@ uri_builder& uri_builder::scheme(std::string str) {
 }
 
 uri_builder& uri_builder::userinfo(std::string str) {
-  impl_->authority.userinfo = std::move(str);
+  return userinfo(std::move(str), std::nullopt);
+}
+
+uri_builder& uri_builder::userinfo(std::string str,
+                                   std::optional<std::string> password) {
```

Comment:
Semantically, it makes no sense to have an overload with two parameters where the second can be disabled again, essentially giving you the same as the overload taking a single argument. If there's a second argument, it should be valid:

```suggestion
uri_builder& uri_builder::userinfo(std::string str, std::string password) {
```

Reply:
Yeah, I was too fixated on differentiating between an empty password and no password that I missed the point here.